pandas/io/html.py

```diff
@@ -554,10 +657,9 @@ def _text_getter(self, obj):
         return obj.text_content()
 
     def _parse_td(self, row):
-        return row.xpath('.//td|.//th')
-
-    def _parse_tr(self, table):
-        return table.xpath('.//tr')
+        # Look for direct children only: the "row" element here may be a
```

Comment:
Conceptually makes sense but is this comment necessary? We aren't enforcing that row be of a certain tag

Reply:
It is nonsensical that `_parse_td()` be called with a `thead` element, because it is nonsensical that a `thead` element contain `td` or `th` children. The `_parse_td` comment says parameter is a DOM `<tr>` node.

It's an ugly hack. I think that nonsense deserves a comment.